gravitee-node-api/src/main/java/io/gravitee/node/api/cache/Cache.java

```diff
@@ -32,107 +32,317 @@
+     */
     default Completable rxPutAll(final Map<? extends K, ? extends V> m) {
         return Completable.fromRunnable(() -> this.putAll(m)).subscribeOn(Schedulers.io());
     }
 
-    V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction);
-
-    default Single<V> rxComputeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
-        return Single.fromCallable(() -> this.computeIfAbsent(key, mappingFunction)).subscribeOn(Schedulers.io());
+    /**
+     * If the specified key is not already associated with a value, attempt to compute the value of the specified key using the given remapping function.
+     * If the remapping function returns <code>null</code>, the mapping is removed.
+     * @param key the key.
+     * @param remappingFunction the function that is executed to associate the new value in case no value was present.
+     *
+     * @return the new value if no value was present or the existing one if it was already present.
+     */
+    V computeIfAbsent(K key, Function<? super K, ? extends V> remappingFunction);
+
+    /**
+     * Reactive method to compute the value of the specified key using the given mapping function if it is not already associated with a value.
+     * If the remapping function returns <code>null</code>, the mapping is removed.
+     * @param key the key.
+     * @param mappingFunction the function that is executed to associate the new value in case no value was present.
+     *
+     * @return a <code>Maybe</code> containing the new value if no value was present or the existing one if it was already present.
+     */
+    default Maybe<V> rxComputeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
+        return Maybe.fromCallable(() -> this.computeIfAbsent(key, mappingFunction)).subscribeOn(Schedulers.io());
     }
 
+    /**
+     * If the specified key is already associated with a value, attempt to compute the value of the specified key using the given mapping function.
+     * If the remapping function returns <code>null</code>, the mapping is removed.
+     * @param key the key.
+     * @param remappingFunction the function that is executed to associate the new value in case a value was present.
+     *
+     * @return the new value if no value was present or the existing one if it was already present.
+     */
     V computeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction);
 
-    default Single<V> rxComputeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
-        return Single.fromCallable(() -> this.computeIfPresent(key, remappingFunction)).subscribeOn(Schedulers.io());
+    /**
+     * Reactive method to compute the value of the specified key using the given mapping function if it is already associated with a value.
+     * If the remapping function returns <code>null</code>, the mapping is removed.
```

Comment:
Looks like the code never removes the value in case of null.

Reply:
Fixed